### PR TITLE
Desktop: Request 2nd factor auth code as text message.

### PR DESCRIPTION
**Pull request by beaucollins**
When the user is prompted for a 2nd factor the form will now display a link that allows them to request an auth code via text message.

Follows the same UX as the Android and iOS apps.

### CODE REVIEW

**Comment by beaucollins** [comment]
@mjangda changed the two booleans to a single state field so it should be ready to go.

File: client/lib/auth-code-request-store/test/auth-code-request-store-test.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import { expect } from 'chai'
+import Dispatcher from 'dispatcher'
+import { default as Store, requestState } from '../index'
+import { actions as ActionTypes } from '../constants'
+
+const debug = require( 'debug' )( 'calypso:comment-store:test' ); //eslint-disable-line no-unused-vars
+//
+// const initialState = {
```
Comment (mjangda) [inline client/lib/auth-code-request-store/test/auth-code-request-store-test.js]: Minor: we can probably remove this commented out code.

File: config/development.json [hunk 2]
```diff
@@ -19,6 +19,8 @@
 		"business_group_id": false
 	},
 	"signup_url": "/start",
+	"desktop_oauth_client_id": "",
+	"desktop_oauth_client_secret": "",
```
Comment (johngodley) [inline config/development.json]: These have been moved to `secrets.json` so we don't need them directly in config files anymore.

**Comment by mjangda** [comment]
Two bits of code that can be removed; otherwise :shipit:

**Comment by beaucollins** [comment]
Looks like the tests require those keys to be present in `development.json`.

**Pull request merged**